### PR TITLE
Remove deprecations

### PR DESCRIPTION
**Pull request by dereuromark**
Fixed deprecations using my upgrade tool

    bin/cake upgrade fixture_casing /path/to/cakephp/migrations/

is the plugin name also uppercase? or only the Fixture?

### CODE REVIEW

**Comment by markstory** [comment]
> is the plugin name also uppercase? or only the Fixture?

Only the fixture class name needs uppercasing.

**Comment by dereuromark** [comment]
Thanks for helping out

> PHP Fatal error:  Cannot use Cake\Core\Plugin as Plugin because the name is already in use

Looks like this fun issue is never going away is it? :)

**Comment by markstory** [comment]
I forgot about that. I will get that fixed too. The only way we could 'solve' this problem without modifying plugin code would be to support another class name for the plugin class.

I think we had discussed using `MigrationsPlugin` but didn't go that way as we assumed people could manage aliasing, and that using `Migrations\Plugin` would provide a more consistent convention.

**Comment by dereuromark** [comment]
Yeah, now its a bit difficult to change without confusing people more.

**Comment by saeideng** [comment]
there are a lot of usage on `Cake\Core\Plugin;` name in this repo :)
I just fixed  one of them yet

**Comment by markstory** [comment]
Looks good to me.

**Comment by dereuromark** [comment]
prefer lowest doesnt like

> PHP Fatal error:  Call to undefined method Migrations\Test\ConfigurationTraitTest::loadPlugins()

File: composer.json [hunk 1]
```diff
@@ -20,12 +20,12 @@
     "require": {
         "php": ">=5.6.0",
         "robmorgan/phinx": "~0.10.3",
-        "cakephp/orm": "^3.6.0",
-        "cakephp/cache": "^3.6.0"
+        "cakephp/orm": "^3.7.0",
+        "cakephp/cache": "^3.7.0"
```
Comment (dereuromark) [inline composer.json]: Sure this was necssary for require?

Comment (ADmad) [reply composer.json]: It's not necessary.

Comment (saeideng) [reply composer.json]: seems yes , it's necessary
https://github.com/cakephp/migrations/blob/1e8074ece648e9001117fff50a43c55b35b9de21/src/Command/CacheClear.php#L61

Comment (ADmad) [reply composer.json]: `Cache::delete()` exits in 3.6 too.

Comment (saeideng) [reply composer.json]: then , we need 3.7 for these :)
https://github.com/cakephp/migrations/pull/378/files
 I know its on core package

Comment (ADmad) [reply composer.json]: I don't understand what you are trying to say.

File: src/Plugin.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+     */
+    protected $name = 'Migrations';
+
+    /**
+     * Override to do nothing.
+     *
+     * @param \Cake\Routing\RouteBuilder $routes The route builder to update.
+     * @return void
+     */
+    public function routes($routes)
```
Comment (ADmad) [inline src/Plugin.php]: Overriding the method is not necessary. Can instead set `$routesEnabled` property to false.

File: composer.json (hunk 1 above)
Comment (ADmad) [reply composer.json]: I don't understand what you are trying to say.

Comment (dereuromark) [reply composer.json]: But in this case it really did install "cakephp/cakephp (3.6.0)"
As such you should only have needed to raise require-dev as the tests need a higher version.
I do see the point of not "knowing" if older versions work, so either way it's not really cool..

File: src/Plugin.php (hunk 2 above)
Comment (ADmad) [inline src/Plugin.php]: Overriding the method is not necessary, instead you can set `$routesEnabled` property to false.

**Comment by ADmad** [comment]
Running the testsuite for this plugin locally is PITA. It pollutes the working directory with numerous generated files.

**Comment by ADmad** [comment]
There ya go, all green while keeping core dependency to 3.6+.

**Comment by ADmad** [comment]
Should the `"minimum-stability": "dev"` in composer.json be removed? I don't think it's required anymore.

**Comment by dereuromark** [comment]
But with 4.x etc, we will need it back again, so best to keep it?

**Comment by ADmad** [comment]
> But with 4.x etc, we will need it back again, so best to keep it?

No we don't. Check bake's 4.x's composer.json for e.g. https://github.com/cakephp/bake/blob/4.x/composer.json

**Comment by dereuromark** [comment]
ok

**Pull request merged**